pkg/apis/core/types.go

```diff
@@ -2125,6 +2133,12 @@ type Lifecycle struct {
 	// or until the termination grace period is reached.
 	// +optional
 	PreStop *Handler
+	// Type describes the startup/shutdown behaviour of the container.
+	// This field is not supported for InitContainers.
+	// Accepted values are "Sidecar".
+	// If unset, the container will obey the normal container lifecycle.
```

Comment:
FYI there's a parallel discussion about the same defaulting problem wrt other fields, so we may change general guidance around this sort of field. The part that I think might be of note for you is that I am proposing that you leave the default as `nil` in versioned API but be explicit with "internal" API.  This PR highlights a complication there, since you have to make teh surrounding struct, too.  Blech.  

FYI only at this point.